cpp/src/arrow/compute/api_scalar.h

```diff
@@ -624,15 +636,20 @@ ARROW_EXPORT
 Result<Datum> Day(const Datum& values, ExecContext* ctx = NULLPTR);
 
 /// \brief DayOfWeek returns number of the day of the week value for each element of
-/// `values`. Week starts on Monday denoted by 0 and ends on Sunday denoted by 6.
+/// `values`. By default week starts on Monday denoted by 0 and ends on Sunday denoted
```

Comment:
Nit, but the `\brief` line should be a one-line summary. Leave an empty line before the more detailed description.

Reply:
Done.